tools/pv_tree_generator/build_tree.py

```diff
@@ -50,39 +50,53 @@ def build_tree_recursive(pos, level, pop_obs_spec, stat_vars,
-            }
-            if sv.se:
-                value_blob['se'] = sv.se
-            # add statistical variables as the child of current node
-            result['cd'].append(value_blob)
-
-            if level <= MAX_LEVEL:
-                # build the branches recursively
-                for child in child_pos:
-                    branch = build_tree_recursive(
-                        child, level + 1, pop_obs_spec, stat_vars, value_ui_node)
-                    if branch['cd']:
-                        if 'cd' not in value_blob:
-                            value_blob['cd'] = []
-                        value_blob['cd'].append(branch)
-                    value_blob['sv_set'] |= branch['sv_set']
-                    del branch['sv_set']
-            value_blob['c'] = len(value_blob['sv_set'])
+            if sv.pv[property_diff] not in child_prop_vals:
+                # if this is the first statsvar of this node, build the node
+                # and corresponding child nodes
+                child_prop_vals.append(sv.pv[property_diff])
+                value_ui_pv = collections.OrderedDict()
+                for prop, val in parent_pv.items():
+                    value_ui_pv[prop] = val
+                value_ui_pv[property_diff] = sv.pv[property_diff]
+                value_ui_node = util.UiNode(
+                    pos, value_ui_pv, False, property_diff)
+                value_blob = {
+                    'populationType': value_ui_node.pop_type,
+                    'sv': [sv.dcid],
+                    'l': text_format.format_title(value_ui_node.text),
+                    't': 'v',
+                    'e': value_ui_node.enum,
+                    'c': 1,
+                    'sv_set': set([sv.dcid]),
+                }
+                if sv.se:
+                    value_blob['se'] = sv.se
+                # add statistical variables as the child of current node
+                result['cd'].append(value_blob)
+
+                if level <= MAX_LEVEL:
+                    # build the branches recursively
+                    for child in child_pos:
+                        branch = build_tree_recursive(
+                            child, level + 1, pop_obs_spec, stat_vars, value_ui_node)
+                        if branch['cd']:
+                            if 'cd' not in value_blob:
+                                value_blob['cd'] = []
+                            value_blob['cd'].append(branch)
+                        value_blob['sv_set'] |= branch['sv_set']
+                        del branch['sv_set']
+                value_blob['c'] = len(value_blob['sv_set'])
+            else:
+                # if this is a duplicated statsVar
+                # append the statsVar to the right node, increase the count by 1
+                for child in result['cd']:
+                    if child['e'] == sv.pv[property_diff]:
+                        child['sv'].append(sv.dcid)
+                        child['c'] += 1
```

Comment:
might not need to increase the count here. the count is more of the UI node rather than the stats var count.

Reply:
done.